# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RDDProcessManager.cpp
# Commit: bf32f5089547
# Full Hash: bf32f5089547afa36393ab6771b45e7ab5ba61ef
# Author: alwu <alwu@mozilla.com>
# Date: 2021-03-03 04:16:49
# Description:
#   Bug 1685666 - check RDDChild before creating content bridge. r=mattwoodrow
#   
#   As the process of creating content process is async, the RDD process might be shutdown before that (eg. crash). So we should check if RDD still exists first.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106827
# ==============================================================================

diff -r 55b0990074bc -r bf32f5089547 dom/media/ipc/RDDProcessManager.cpp
--- a/dom/media/ipc/RDDProcessManager.cpp	Tue Mar 02 04:36:57 2021 +0000
+++ b/dom/media/ipc/RDDProcessManager.cpp	Tue Mar 02 06:18:36 2021 +0000
@@ -208,6 +208,11 @@
   return !!mProcess && !mRDDChild;
 }
 
+bool RDDProcessManager::IsRDDProcessDestroyed() const {
+  MOZ_ASSERT(NS_IsMainThread());
+  return !mRDDChild && !mProcess;
+}
+
 void RDDProcessManager::OnProcessUnexpectedShutdown(RDDProcessHost* aHost) {
   MOZ_ASSERT(NS_IsMainThread());
   MOZ_ASSERT(mProcess && mProcess == aHost);
@@ -261,6 +266,12 @@
     ipc::Endpoint<PRemoteDecoderManagerChild>* aOutRemoteDecoderManager) {
   MOZ_ASSERT(NS_IsMainThread());
 
+  if (IsRDDProcessDestroyed()) {
+    MOZ_LOG(sPDMLog, LogLevel::Debug,
+            ("RDD shutdown before creating content bridge"));
+    return false;
+  }
+
   ipc::Endpoint<PRemoteDecoderManagerParent> parentPipe;
   ipc::Endpoint<PRemoteDecoderManagerChild> childPipe;
 